site/docs/v1/tech/apis/webauthn.adoc

```diff
@@ -135,4 +144,140 @@ include::docs/v1/tech/apis/_import-webauthn-request-body.adoc[]
+--
+
+include::docs/v1/tech/apis/_x-fusionauth-tenant-id-header-scoped-operation.adoc[]
+
+include::docs/v1/tech/apis/_webauthn-register-complete-request-body.adoc[]
+
+=== Response
+
+:omit_missing_header:
+include::docs/v1/tech/apis/_standard-post-response-codes.adoc[]
+:omit_missing_header!:
+
+include::docs/v1/tech/apis/_webauthn-response-body.adoc[]
+
+== Start a WebAuthn Passkey Assertion
+
+This API is used to start a WebAuthn authentication ceremony by providing some details about the current user and the new passkey. The response contains the request options that will be passed to the authenticator on the client device, including a one-time challenge unique to the current ceremony.
+
+=== Request
+
+[.api-authentication]
+link:/docs/v1/tech/apis/authentication#api-key-authentication[icon:lock[type=fas]] Start a WebAuthn Authentication Ceremony and Retrieve Passkey Request Options
+[.endpoint]
+.URI
+--
+[method]#POST# [uri]#/api/webauthn/start#
+--
+
+include::docs/v1/tech/apis/_x-fusionauth-tenant-id-header-scoped-operation.adoc[]
+
+include::docs/v1/tech/apis/_webauthn-authenticate-start-request-body.adoc[]
+
+=== Response
+
+The response for this API contains the WebAuthn passkey request options that are passed to the WebAuthn API on the client. A few fields on the API response are returned as a base64url-encoded string and need to be converted to an `ArrayBuffer` in JavaScript before being passed to the WebAuthn client API.
+
+include::docs/v1/tech/apis/_standard-post-response-codes.adoc[]
+
+include::docs/v1/tech/apis/_webauthn-authenticate-start-response-body.adoc[]
+
+== Complete a WebAuthn Passkey Authentication
+This API is used to complete a WebAuthn authentication ceremony by providing the values returned from the WebAuthn client API call. The API will validate the request against configured passkey requirements for the workflow and the one-time challenge generated and returned by <<Start a WebAuthn Passkey Assertion>>.
+
+=== Request
+
+[.api-authentication]
+link:/docs/v1/tech/apis/authentication#api-key-authentication[icon:unlock[type=fas]] Complete a WebAuthn Authentication Ceremony
+[.endpoint]
+.URI
+--
+[method]#POST# [uri]#/api/webauthn/login#
+--
+
+include::docs/v1/tech/apis/_x-fusionauth-tenant-id-header-scoped-operation.adoc[]
+
+==== Request Cookies
+
+[.api]
+[field]#fusionauth.trust# [type]#[String]# [optional]#Optional#::
+The Multi-Factor Trust identifier returned by the Multi-Factor Login API response. This value be may be provided to bypass the Multi-Factor challenge when a User has Multi-Factor enabled. When this cookie exists on the request it will take precedence over the [field]#twoFactorTrustId# if provided in the request body.
+
+include::docs/v1/tech/apis/_webauthn-authenticate-complete-request-body.adoc[]
+
+=== Response
+The response for this API contains the User object.
+
+include::docs/v1/tech/apis/_login-response-codes.adoc[]
+
+:login_response:
```

Comment:
Is a 503 response possible?

Reply:
I doubt it. Is there another include that doesn't have that status code? It seems like every one of the standard API response code includes has the 503.

The only places I see it used are for device grant and failing to generate an external Id. And then the message that comes with the 503 everywhere doesn't make sense.

Reply:
My understanding is that a 503 response is only possible when a search index is used (so looking up anything in elasticsearch). Any prime action where `service-unavailable` is returned or the ServiceUnavailableException is thrown. But that may be a bigger fish to fry because I'm not quite sure that is correct.